crates/outbound-http/src/host_impl.rs

```diff
@@ -63,9 +66,14 @@ impl outbound_http::Host for OutboundHttp {
 
             let req_url = reqwest::Url::parse(&abs_url).map_err(|_| HttpError::InvalidUrl)?;
 
-            let headers = request_headers(req.headers).map_err(|_| HttpError::RuntimeError)?;
+            let mut headers = request_headers(req.headers).map_err(|_| HttpError::RuntimeError)?;
             let body = req.body.unwrap_or_default().to_vec();
 
+            // if no content length, set it
+            if !body.is_empty() && !headers.contains_key(CONTENT_LENGTH) && !headers.contains_key(TRANSFER_ENCODING){
+                headers.insert("content-length", body.len().into());
+            }
```

Comment:
Initially I thought this was wrong because the body size could change for streaming bodies, but then realized that this interface doesn't support those. I think at least this should contain a comment to that effect, but IIUC it could just use the same approach as in the wasi-http implementation below, right?

And I think in both cases, you don't even need to extract the body: you should be able to use `if let Some(size) = req.size_hint().exact()`, which I think should just not match (because `exact()` will return `None` if no body is present, or the final size isn't known.